core/iwasm/libraries/libc-wasi/libc_wasi.cmake

```diff
@@ -5,6 +5,12 @@ set (LIBC_WASI_DIR ${CMAKE_CURRENT_LIST_DIR})
 
 add_definitions (-DWASM_ENABLE_LIBC_WASI=1)
 
+if (WAMR_BUILD_ENABLE_LIBC_WASI_SYNC_FLAGS EQUAL 1)
+    add_definitions (-DWASM_ENABLE_LIBC_WASI_SYNC_FLAGS=1)
```

Comment:
As suggested by yamt, how about `-DCONFIG_HAS_LIBC_WASI_SYNC_FLAGS=1` instead? And in ssp_config.h, add:
```C
#ifndef CONFIG_HAS_LIBC_WASI_SYNC_FLAGS
#define CONFIG_HAS_LIBC_WASI_SYNC_FLAGS 0
#endif
```